index.js

```diff
@@ -90,14 +90,23 @@ IFrameClass.prototype._injectIFrame = function(iframe, options) {
 
   options || (options = {});
   options.src && (this.targetOrigin = options.src);
-  $iframe.attr('src', this.targetOrigin);
+  var iframeId = Random.guid();
+  $iframe.attr('name', iframeId);
+
+  $('body').append($iframe);
   this._isIFrameInjected = true;
-  $iframe.on('load', function() {
+
+  $iframe.on('load', function () {
     window.clearTimeout(this._iframeLoadTimeoutIndex);
     deferred.resolve($iframe.get(0), $iframe);
+    $iframe.removeAttr('name')
   });
-  $('body').append($iframe);
-  this._iframeLoadTimeoutIndex = window.setTimeout(function() {
+
+  iframe = window.open(this.targetOrigin, iframeId)
+
+  this.childWindow = iframe
+
```

Comment:
What is the benefit of this change?

Reply:
I use this to compare with the event.source

Reply:
We can get iframe element ref by using `$iframe.get(0)` instead.

Reply:
both are same, I prefer the one returned from the opener method itself, since it directly returns the window.

Reply:
Reverted.